core/modules/util/MultiError.h

```diff
@@ -0,0 +1,71 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2014 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+/**
+* @file
+*
+* @ingroup util
+*
+* @brief Implement a generic error container for Qserv
+*
+* @author Fabrice Jammes, IN2P3/SLAC
+*/
+
+#ifndef LSST_QSERV_UTIL_ERROR_H
+#define LSST_QSERV_UTIL_ERRORCONTAINER_H
+
+// System headers
+#include <exception>
+#include <ostream>
+#include <vector>
+
+// Qserv headers
+#include "util/Error.h"
+
+namespace lsst {
+namespace qserv {
+namespace util {
+
+/** @class
+ * @brief Implement a generic error container for Qserv
+ *
+ * Store Qserv errors in a throwable vector.
+ * util::Error operator << is used for output.
+ *
+ */
+class MultiError: public std::exception, public std::vector<Error> {
```

Comment:
Uh oh...  It's is generally considered bad practice to derive from standard containers (for example, historically, std::vector<> didn't have a virtual destructor -- is this still the case?)  You might try googling "derive from standard containers" and read up on reasons why.

I'd suggest composition here instead of inheritance -- try having MultiError hold a vector internally, then implement your own begin(), end() methods that return iterators to the underlying container.

std::exception is explicitly designed to be used as a base class, so that one is fine.